src/EmployeesAdministrationApp.java

```diff
@@ -85,42 +82,48 @@ private static void addEmployee(Object employee, String message) {
 	}
 
 	private static void writeJson(Object obj) {
-		String employeesJsonFile = "employees1.json";
-		JSONParser parser = new JSONParser();
+		String employeesJsonFile = "employees.json";
+		JSONArray allEmployeesList = new JSONArray();
+		allEmployeesList = readFromFile(employeesJsonFile, allEmployeesList);
+		Method[] methods = obj.getClass().getMethods();
+		Map<String, Object> employeeDetails = new HashMap<String, Object>();
+		for (Method method : methods) {
+			if (method.getName().startsWith("get") && !method.getName().equals("getClass")) {
+				try {
+					employeeDetails.put(method.getName().replace("get", ""), method.invoke(obj));
+				} catch (Exception e) {
+					e.printStackTrace();
+				}
+			}
+		}
+		Map<String, Map<String, Object>> employeeDetailsJson = new HashMap<>();
+		employeeDetailsJson.put("employee", employeeDetails);
+		JSONObject employeeDetailsJsonObject = new JSONObject(employeeDetailsJson);
+		allEmployeesList.add(employeeDetailsJsonObject);
+		try {
+			FileWriter file = new FileWriter(employeesJsonFile);
+			file.write(allEmployeesList.toJSONString());
+			file.flush();
+		} catch (IOException e) {
+			System.out.println("There was a problem in writing data to this file");
+		}
+		System.out.println("You have added a new " + obj.getClass().getName() + " with: " + obj + "\nin your "
+				+ employeesJsonFile + " document");
+	}
+
+	private static JSONArray readFromFile(String employeesJsonFile, JSONArray allEmployeesList) {
```

Comment:
you don't need the second parameter, since you are returning it: JSONArray allEmployeesList

Reply:
I deleted this method since now the project has a load file option